### ISSUE TITLE
[Feature]: re-enable gpu-support

### ISSUE THREAD

**Issue by kitsudaiki** [kitsudaiki/ainari#325]
### Description

At the moment the gpu-support is disabled, because there were some changes at the core while there was at the same time no gpu available to test the update in the CUDA-code. 

### Acceptance Criteria

- enable gpu-support again
- update CUDA-kernel and the code for the data-transfer from and to the gpu for the current state of the core

### Additional Information

- #276 
- #307 

### Blocked by

- #370

**Comment by kitsudaiki** [comment]
Update of the cuda-code so far is incomplete and will be set on hold for a while. I had an idea for a new update of the core-structure and the cuda kernel is not abosult necessary at the moment. The current updates were merged anyway, because there was also some cleanup like the split of the kernel-file. 
Will be set to on-hold and planned new for version 0.6.0

**Comment by kitsudaiki** [comment]
blocked by issue #370

**Comment by kitsudaiki** [comment]
cuda-kernel completely disabled form now in context of issue https://github.com/kitsudaiki/Hanami/issues/377

Need to collect new ideas before I make a new attempt to add gpu support to the project.

**Comment by kitsudaiki** [comment]
updated code, but while testing I ran into a weird error: After the second cycle the worker-task disappeared from the worker-queue of the cuda-worker-thread, right after the task was plased into the queue. Seems like a memory-corruption, but the recently added ASan memory-check doesn't work with the CUDA-code (doesn't detect a CUDA-gpu when ASan is activated). I don't wanted to waste more time at the moment. Will try at again another time. So the CUDA-kernel is still disabled within the Code and the issue here remains as on-hold

**Comment by kitsudaiki** [comment]
deprecated because of https://github.com/kitsudaiki/ainari/issues/906
